src/coreclr/vm/interpexec.cpp

```diff
@@ -1126,6 +1185,9 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
     }
 
 EXIT_FRAME:
+
+    // Interpreter-TODO: Don't run PopInfo on the main return path, Add RET_LOCALLOC instead
+    pThreadContext->frameDataAllocator.PopInfo(pFrame);
```

Comment:
@janvorli Note that this is currently expected to be called for every single method frame that exits (even if via an exception), so this might leak depending on the EH approach. Do you think we should leave it as is or tweak it now ? A simple solution, that would unfortunately free the memory later at times, would be free during alloc. We would pop all entries for frames above the frame that we are pushing (aka for frames that are no longer valid, since the frame pointers are monotonic)

Reply:
Let's leave it this way for this PR, I'll take care of it in my EH PR.